File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        
        for(char z : s.toCharArray()){
            
            if(Character.isLetterOrDigit(z)){
                sb.append(Character.toLowerCase(z));
            }
        }
        
        String newStr = sb.toString();
        int left = 0;
        int right = newStr.length() - 1;
        
        while(left < right){
            if(newStr.charAt(left) != newStr.charAt(right)){
                return false;
            }
            
            left++;
            right--;
        }
        
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
function isPalindrome(s){
    let str1 = s.toLowerCase().replace(/[^a-z0-9]/g, '');
    console.log(str1)

    let str = str1.split('')
    console.log(str)

    let i=0;
    let j = str.length - 1;

    while(i < j){
        if(str[i] !== str[j]){
            return false;
        }

        i++;
        j--;
    }


    return true;
};